pkg/kubelet/container.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"hash/adler32"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/rocket"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/golang/glog"
+)
+
+const defaultEndpoint = "/home/yifan/gopher/src/github.com/coreos/rocket/bin/rkt"
+
+var rkt *rocket.RocketRuntime
+
+func init() {
+	var err error
+	rkt, err = rocket.NewRocketRuntime(defaultEndpoint)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// FindContainerInPod finds the container in the pod.
+func FindContainerInPod(container *api.Container, pod *api.Pod) (*api.Container, bool) {
+	for _, c := range pod.Spec.Containers {
+		if c.Name == container.Name {
+			return &c, true
+		}
+	}
+	return nil, false
+}
+
+// RunContainerInPod starts a container in the given pod.
+func RunContainerInPod(container *api.Container, pod *api.Pod) error {
+	// Kill the pod and restart.
+	if err := rkt.KillPod(pod); err != nil {
+		return err
+	}
+
+	// Update the pod and start it.
+	pod.Spec.Containers = append(pod.Spec.Containers, *container)
+	boundPod := &api.BoundPod{pod.TypeMeta, pod.ObjectMeta, pod.Spec}
+	if err := rkt.RunPod(boundPod); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Help function, remove the container in the given pod.
+func removeContainer(pod *api.Pod, container api.Container) {
```

Comment:
This method signature is the only one that swaps the ordering of `pod` and `container`. Is there a good reason for that?